### PR TITLE
Change Log should overwrite the default log name

### PR DESCRIPTION
**Pull request by craig-cogdill**
Logs weren't rotating over to the new name if you used ```changeLogName()```, they were rolling over to the original default name.

Added a test for this case as well.

### CODE REVIEW

File: logrotate/src/LogRotateHelper.ipp [hunk 1]
```diff
@@ -267,11 +267,13 @@ std::string LogRotateHelper::changeLogFile(const std::string& directory, const s
     std::string file_name = new_name;
     if (file_name.empty()) {
         std::cout << "no filename" << std::endl;
-       file_name = createLogFileName(log_prefix_backup_);
+        file_name = log_prefix_backup_;
+    } else {
+        log_prefix_backup_ = file_name;
```
Comment (KjellKod) [inline logrotate/src/LogRotateHelper.ipp]: This does not make sense.  The log prefix backup should not be changed until the file name is validated.

Comment (KjellKod) [inline logrotate/src/LogRotateHelper.ipp]: After line 281 we know the file name is good. After this we can change the default backup

File: logrotate/test/RotateFileTest.cpp [hunk 2]
```diff
@@ -77,8 +77,27 @@ TEST_F(RotateFileTest, setMaxLogSize) {
 
    exists = Exists(content, first_message_in_new_log);
    EXPECT_TRUE(exists) << "\n\tcontent:" << content << "-\n\tentry: " << gone;
+}
+
+TEST_F(RotateFileTest, setMaxLogSizeAndRotate) {
```
Comment (KjellKod) [inline logrotate/test/RotateFileTest.cpp]: Can you add a unit test for empty file name since that logic is slightly changed?

File: logrotate/src/LogRotateHelper.ipp (hunk 1 above)
Comment (craig-cogdill) [inline logrotate/src/LogRotateHelper.ipp]: Okay

**Comment by KjellKod** [comment]
No need for the empty test since you changed it

**Comment by KjellKod** [comment]
:+1:
